### PR TITLE
package: Upgrade dependencies to resolve security issues.

### PR DESCRIPTION
**Pull request by Henry8192**
# Description
https://github.com/y-scope/clp/security/dependabot reports issue on webui and log-viewer-webui. This PR addresses these issues by simply either running `npm i` or `npm audit fix`.

# Validation performed
In webui and log-viewer-webui folder, run `npm i` and observe 0 vulnerabilities.

## Summary by CodeRabbit

- **New Features**
	- Updated dependencies for improved performance and stability.

- **Bug Fixes**
	- Upgraded `webpack` and `webpack-dev-server` to enhance build processes.
	- Updated `meteor-node-stubs` for better compatibility.

- **Chores**
	- Modified version constraints for `setuptools` to allow newer versions.

### CODE REVIEW

**Comment by junhaoliao** [comment]
For the PR title, how about:
```
webui & log-viewer-webui: Resolve dependency security issues.
```

File: components/log-viewer-webui/client/package.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "mini-css-extract-plugin": "^2.9.0",
     "react-refresh": "^0.14.2",
     "style-loader": "^4.0.0",
-    "webpack": "^5.92.1",
+    "webpack": "^5.94.0",
```
Comment (junhaoliao) [inline components/log-viewer-webui/client/package.json:27]: There seems to be a newer version now.

```suggestion
    "webpack": "^5.95.0",
```

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 pip~=23.3
 poetry~=1.7
-setuptools~=69.0
+setuptools>=70.0.0
```
Comment (junhaoliao) [inline requirements.txt:3]: Is being more explicit better?
```suggestion
setuptools~=75.2.0
```

**Comment by junhaoliao** [comment]
Let's fix the code at https://github.com/y-scope/clp/blob/2c99375202b9f49991f13503c936854862bce8df/components/log-viewer-webui/server/src/DbManager.js
Line 5: 
```
import {encode as msgpackEncode} from "@msgpack/msgpack";
```
...
Line 116:
```
                    Buffer.from(msgpackEncode(jobConfig)),

```

**Comment by junhaoliao** [comment]
> > For the PR title, how about:
> > ```
> > Resolve dependency security issues.
> > ```
> 
> What do you think of this? `webui & log-viewer-webui: resolve security issues caused by dependencies`

Since we also updated `requirements.txt`, it's no longer specific to `webui & log-viewer-webui`

File: components/log-viewer-webui/server/src/DbManager.js [hunk 3]
```diff
@@ -113,7 +113,7 @@ class DbManager {
                 `INSERT INTO ${this.#queryJobsTableName} (job_config, type)
              VALUES (?, ?)`,
                 [
-                    Buffer.from(msgpack.encode(jobConfig)),
+                    Buffer.from(encode(jobConfig)),
```
[the remark this answers is not in the document]
Comment (junhaoliao) [reply components/log-viewer-webui/server/src/DbManager.js:116]: @Henry8192 can we address this before we merge the PR?

File: components/log-viewer-webui/client/package-lock.json [hunk 4]
```diff
@@ -10559,23 +10772,21 @@
         "colorette": "^2.0.10",
         "compression": "^1.7.4",
         "connect-history-api-fallback": "^2.0.0",
-        "default-gateway": "^6.0.3",
-        "express": "^4.17.3",
+        "express": "^4.19.2",
```
Comment (kirkrodrigues) [inline components/log-viewer-webui/client/package-lock.json:10775]: This version still seems to be [vulnerable](https://github.com/y-scope/clp/security/dependabot/21).

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: components/log-viewer-webui/client/package-lock.json (hunk 4 above)
Comment (Henry8192) [reply components/log-viewer-webui/client/package-lock.json:10775]: The version that's installed is actually newer. 4.19.2 is actually just the minimum version that they require.

Comment (kirkrodrigues) [reply components/log-viewer-webui/client/package-lock.json:10775]: Right, my mistake.

**Review by kirkrodrigues — APPROVED** [review]
For the PR title, how about:

```
package: Upgrade dependencies to resolve security issues.
```

**Pull request merged**